ext/LegendSpecFitsRecipesBaseExt.jl

```diff
@@ -446,72 +446,67 @@ end
-    thickness_scaling := 1.0
+    layout := (2, 1)
+    size := (600, 700)
+    framestyle := :semi
+    grid := false 
+    left_margin --> (5, :mm)
+    right_margin --> (5, :mm)
+    foreground_color_legend := :silver
+    background_color_legend := :white
     xtickfontsize := 12
     xlabelfontsize := 14
     ylabelfontsize := 14
     ytickfontsize := 12
-    legendfontsize := 10
-    size := (1000, 300)
-    margin := (8, :mm)
-    @series begin
-        seriestype := :histogram2d
-        bins := (ustrip.(unit(first(report_ctc.e_peak)), minimum(report_ctc.e_peak):report_ctc.bin_width:maximum(report_ctc.e_peak)), quantile(report_ctc.qdrift_peak, 0.01):report_ctc.bin_width_qdrift:quantile(report_ctc.qdrift_peak, 0.99))
-        color := :inferno
-        xlabel := "Energy"
-        ylabel := "QDrift"
-        title := "Before Correction"
-        titlelocation := (0.5, 1.1)
-        xlims := (2600, 2630)
-        ylims := (0, quantile(report_ctc.qdrift_peak, 0.99))
-        yformatter := :plain
-        legend := :none
-        colorbar_scale := :log10
+    legendfontsize := 12
+    xl = (2590, 2641)
+    hist_bins = range(xl[1], xl[2], 100)
+    @series begin
+        seriestype := :stephist 
+        fill := true
+        color := :darkgrey
+        label := "Before correction"
+        legend := :topleft
         subplot := 1
-        report_ctc.e_peak, report_ctc.qdrift_peak
+        bins := hist_bins
+        report_ctc.e_peak
     end
     @series begin
-        seriestype := :histogram2d
-        bins := (ustrip.(unit(first(report_ctc.e_peak)), minimum(report_ctc.e_peak):report_ctc.bin_width:maximum(report_ctc.e_peak)), quantile(report_ctc.qdrift_peak, 0.01):report_ctc.bin_width_qdrift:quantile(report_ctc.qdrift_peak, 0.99))
-        color := :magma
+        seriestype := :stephist 
+        fill := true
+        alpha := 0.5
+        color := :purple
+        label := "After correction"
+        legend := :topleft
+        subplot := 1
+        ylims := (0, :auto)
+        xlims := xl
         xlabel := "Energy"
-        ylabel := "QDrift"
-        title := "After Correction"
-        xlims := (2600, 2630)
-        titlelocation := (0.5, 1.1)
-        xlims := (2600, 2630)
-        ylims := (0, quantile(report_ctc.qdrift_peak, 0.99))
-        yformatter := :plain
-        legend := :none
-        colorbar_scale := :log10
-        subplot := 3
-        report_ctc.e_ctc, report_ctc.qdrift_peak
+        bins := hist_bins
+        report_ctc.e_ctc
     end
     @series begin
-        seriestype := :stepbins
-        color := :red
-        label := "Before CTC"
-        xlabel := "Energy (keV)"
-        ylabel := "Counts"
-        yscale := :log10
+        seriestype := :line
         subplot := 2
-        report_ctc.h_before
+        c := :binary
+        colorbar := :none
+        fill := true
+        label := "Before correction"
+        BAT.kde((ustrip(report_ctc.e_peak), report_ctc.qdrift_peak ./ maximum(report_ctc.qdrift_peak)))
```

Comment:
The function `kde` is not defined in `BAT`, but in `KernelDensity`, which is exported into `BAT`.
It would be cleaner to remove the `BAT` dependency and replace it with `KernelDensity` for version control.

Reply:
We'd like to get rid of KernelDensity in BAT, ideally ... :-)